src/ListContainer.hpp

```diff
@@ -39,6 +39,65 @@ struct ipl_rangestruct {
 }
 #endif
 
+// class for linking IPs to filter groups, complete with comparison operators
+// allowing standard C++ sort to work
+class ipmap
+{
+public:
+    ipmap(uint32_t a, String g)
+    {
+        addr = a;
+        group = g;
+    };
+    uint32_t addr;
+    String group;
+    int operator<(const ipmap &a) const
+    {
+        return addr < a.addr;
+    };
+    int operator<(const uint32_t &a) const
+    {
+        return addr < a;
+    };
+    int operator==(const uint32_t &a) const
+    {
+        return a == addr;
+    };
+};
+
+class datamap
+{
+public:
+    datamap(String &k, String &g){
+        key = k;
```

Comment:
Codacy found an issue: [Variable 'key' is assigned in constructor body. Consider performing initialization in initialization list.](https://app.codacy.com/manual/numsys/e2guardian/pullRequest?prid=6238856)